### ISSUE TITLE
Detailed test and analysis of Maqetta-to-Eclipse workflow

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#1548]
@childsb 

We need to do detailed testing to verify that we have addressed all issues with Eclipse-compatible projects

**Comment by JonFerraiolo** [comment]
Reassigning to @aerwin , who is already doing this analysis.

**Comment by aerwin** [comment]
More details on what's been going on with this issue can be found in Bugzilla 7894.

**Comment by jhpedemonte** [comment]
Is there anything left to do here?  If test/analysis is done, then close this issue and open new ones for remaining bugs.

**Comment by aerwin** [comment]
@jhpedemonte Yes, been kind of waiting on RAD Beta 2 (https://www14.software.ibm.com/iwm/web/cc/earlyprograms/websphere/wasv8na/index.shtml) to run through some things. It just came out this week, so hoped to run a few quick tests by end of week.

**Comment by aerwin** [comment]
A few weeks back I did some work with Beta 1 of the Web Developer Tools (WDT), but recently the March refresh of the  beta became available:

http://marketplace.eclipse.org/node/169799

I downloaded the zip file with the WDT plugins and installed into Eclipse (as a local update site):

http://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/V8.5/wdtupdate/wdt-update-site_8.5.0.WDT85-I20120217_1146.zip

Here are some notes from the most recent round of testing:
- WDT still doesn't work with the official release of Dojo 1.7.1. Unfortunately, a workaround developed with the WDT team no longer works (e.g., using the "trick" involving remote CDN settings).
  - I'll follow-up with WDT team if this will be fixed and determine if we'll still be able to recommend using the workaround.
- Can't drag widgets from palette into Maqetta generated HTML files in WDT Rich Page Editor.
  - This was also true in the last beta, but was supposed to have been fixed. I'll follow-up with WDT team.
- Made some tweaks to settings files in Maqetta generate project (based on comparison with settings files in WDT-generated Dojo-enabled project). See commit above (3c34e4fe052adb5a90bb9b317e5555042a444f6d).

**Comment by aerwin** [comment]
Moving to M6 so we keep tracking this over time as new WDT builds become available...

**Comment by aerwin** [comment]
@JonFerraiolo @jhpedemonte 

As I wrote about 10 days ago, I didn't have a lot of success with the the March refresh of tooling. Yesterday, I got hold of an internal development driver of RAD (from Mar. 29) and did some more analysis.

Notes:
- The workaround developed with Beta 1 that started failing in Beta 2 continues to not work. So, I've come up with an alternative (that I think is probably better anyway) to have RAD copy the "right" version of Dojo into the imported project. I've reworked the Wiki page that I authored a few weeks ago to reflect this new approach.
- The previously identified bug where Dojo widgets could not be dropped onto Maqetta-generated HTML files has been addressed.
- Made a couple more tweaks to Eclipse settings files generated to Maqetta to match the settings files found in the dev driver of RAD for client-side only, Dojo-enabled web projects.

I've also done some additional sanity testing with this RAD driver in the areas below. 
- Opened a mobile web page in Rich Page Editor and successfully rendered it in the RAD mobile previewer in Firefox. I was also able to change devices in the previewer and see the theme change according to the device.
- Dragged Dojo widgets from Rich Page Editor palette onto Maqetta-generated generated desktop and mobile HTML files and saw preview and source updated in RAD.
- Opened an HTML page with a custom theme in the Rich Page Editor and see the theme is reflected.
- Opened an HTML page with desktip lo-fi sketch and confirmed widgets rendered in absolute positions with the sketch theme.

This certainly doesn't exhaustively test that everything possible in Maqetta gets properly reflected after importing into RAD. But, I think things are working well enough that we can close this ticket. And, I'm pretty well convinced any issues at this point would more than likely be RAD bugs. 

However, I will open a placeholder task to make sure we occasionally come back and test new RAD drivers.
